snapcraft/cli.py

```diff
@@ -34,31 +36,61 @@
 from . import commands
 from .legacy_cli import run_legacy
 
+
+@dataclass
+class CommandGroup:
```

Comment:
did you not want to use `craft_cli.CommandGroup`?

Reply:
I'm passing it [here](https://github.com/canonical/snapcraft/pull/4830/files#diff-92e6477c380125a0d88c5b0dfccd297e956c7127163fc7ab76ba1bee78f16b9fR275) as well, which [takes](https://github.com/canonical/craft-application/blob/e898740ef34686e6a262e705f0521c62e23036ee/craft_application/application.py#L193) takes just a name and a list of commands.

Reply:
I'd rather fix that upstream

Reply:
As in this?

```diff
 add_command_group(
-        self, name: str, commands: list[type[craft_cli.BaseCommand]]
+        self, craft_cli.CommandGroup
    ) -> None:
```

If so, I can make a task for that in craft-application.

Reply:
We could do that (which is a breaking change), or we can change `add_command_group` to simply take any Sequence rather than list specifically, which will allow using craft_cli's CommandGroup and just writing:

```
add_command_group(group.name, group.commands)
```

Reply:
https://github.com/canonical/craft-application/issues/357